web/elm/src/Dashboard/Dashboard.elm

```diff
@@ -1161,29 +1173,36 @@ pipelinesView session params =
                 , dashboardView = params.dashboardView
                 , favoritedPipelines = session.favoritedPipelines
                 }
-                |> List.sortWith (Group.ordering session)
+                |> Dict.toList
+                |> List.sortWith (Ordering.byFieldWith (Group.ordering session) Tuple.first)
+
+        cardsByTeam =
+            filteredPipelinesByTeam
+                |> List.map (\( team, ps ) -> ( team, ps |> List.map PipelineCard ))
```

Comment:
ugh... I'm not sure if I understand the two maps here tbh...
So the `filteredPipelinesByTeams` returns all the pipelines and pass them into the map, where the actual function `(\( team, ps ) -> ( team, ps |> List.map PipelineCard ))` is a mapping from a tuple of `(team, ps)` to the  function`PipelineCard`?

I'm confused about why we need this tuple here... it seems like PipelineCard only needs a pipeline?

Reply:
Maybe outlining the types explicitly would make it more clear.

We have

```
filteredPipelinesByTeam : List (String, List Pipeline)
```

and we want

```
cardsByTeam : List (String, List Card)
```

so the first `map` converts the outer `List` (of tuples), and the second `map` converts the inner `List` within each tuple (of `Pipelines`). The point of the tuple is just so we can render each team's cards individually